core/src/trezor/ui/layouts/tr/__init__.py

```diff
@@ -1,45 +1,192 @@
-from typing import TYPE_CHECKING, Sequence
+from typing import TYPE_CHECKING
 
-from trezor import io, log, loop, ui, wire, workflow
+from trezor import io, loop, ui
 from trezor.enums import ButtonRequestType
-from trezor.utils import DISABLE_ANIMATION
+from trezor.wire import ActionCancelled
 
 import trezorui2
 
 from ..common import button_request, interact
 
 if TYPE_CHECKING:
-    from typing import Any, NoReturn, Type
+    from typing import Any, NoReturn, Awaitable, Iterable, Sequence, TypeVar
 
-    ExceptionType = BaseException | Type[BaseException]
+    from trezor.wire import GenericContext, Context
+    from ..common import PropertyType, ExceptionType
 
+    T = TypeVar("T")
 
-if __debug__:
-    trezorui2.disable_animation(bool(DISABLE_ANIMATION))
+
+CONFIRMED = trezorui2.CONFIRMED
+CANCELLED = trezorui2.CANCELLED
+INFO = trezorui2.INFO
+
+BR_TYPE_OTHER = ButtonRequestType.Other  # global_import_cache
 
 
-def is_confirmed(x: Any) -> bool:
-    return x is trezorui2.CONFIRMED
+if __debug__:
+    from trezor.utils import DISABLE_ANIMATION
+
+    trezorui2.disable_animation(bool(DISABLE_ANIMATION))
 
 
 class RustLayout(ui.Layout):
```

Comment:
Eventually we'll need to merge this with the implementation for model T. Not strictly in this PR, probably as part of #2634.